lib/json-schema/util/uri.rb

```diff
@@ -11,6 +16,19 @@ def self.normalized_uri(uri)
         end
         uri
       end
+
+      # This cache creates a race condition in multithreaded environments
+      # This is OK because the worst case result of the race is
+      # a cache miss
+      def self.normalized_uri(uri)
+        instance.normalize_cache ||= {}
+        instance.normalize_cache[uri] ||= instance.normalized_uri(uri)
```

Comment:
One last question - should the cache key here be "uri.to_s"? (As it is for the parse cache?)